useful-scripts/misc-test-utils/src/modules/subject/upload/index.ts

```diff
@@ -0,0 +1,226 @@
+/* eslint-disable */
+import axios from 'axios';
+import 'dotenv-safe/config';
+import { v4 } from 'uuid';
+import fs from 'fs';
+import StreamZip from 'node-stream-zip';
+import globby from 'globby';
+import rimraf from 'rimraf';
+import FormData from 'form-data';
+import ZipDirectory from '../../../utils/zipDirectory';
+import Sleep from '../../../utils/Sleep';
+import { SubjectArray } from '../../../interfaces/SubjectArray';
+import errorHandler from '../../../utils/errorHandler';
+import chalk from 'chalk';
+
+// disable insecure warnings
+process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
+const { ADMIN_URL, JWT_TOKEN } = process.env;
+const cwd = process.cwd();
+
+const extractZip = async () => {
+  const zippedFile = await globby(`${cwd}/*.zip`);
+  const cleanZipFile = zippedFile[0];
+  const archive = fs.existsSync(zippedFile[0]);
+  if (archive) {
+    const zip = new StreamZip.async({ file: cleanZipFile });
+    const count = await zip.extract(null, `${cwd}/test-files`);
+    console.log(chalk.green(`Extracted ${count} entries to test-files`));
+    await zip.close();
+  }
+};
+
+const renameFiles = async () => {
+  // define a random id to assign to the renamed data & meta file...
+  const randomId = Math.floor(Math.random() * 1000000);
+
+  // rename csv file..
+  const csvGlob1 = await globby(`${cwd}/test-files/*.csv`);
+
+  const csvGlob2 = csvGlob1[0];
+
+  fs.rename(`${csvGlob2}`, `${cwd}/test-files/testfile-${randomId}.csv`, e =>
+    e ? console.error(chalk.red(e)) : '',
+  );
+
+  // rename the meta file...
+  const metaGlob = await globby(`${cwd}/test-files/*.meta.csv`);
+  const metaGlob2 = metaGlob[0];
+
+  fs.rename(
+    `${metaGlob2}`,
+    `${cwd}/test-files/testfile-${randomId}.meta.csv`,
+    e => (e ? console.error(chalk.red(e)) : ''),
+  );
+};
+
+const addSubject = async (releaseId: string) => {
+  const glob = await globby(`${cwd}/zip-files/*.zip`);
+
+  // potential issue here with how windows paths are formatted in git bash
+  if (glob !== undefined) {
+    const oldPath =
+      process.cwd() +
+      '/zip-files/' +
+      fs.readdirSync(process.cwd() + '/zip-files')[0];
+
+    const newPath =
+      process.cwd() + '/' + fs.readdirSync(process.cwd() + '/zip-files')[0];
```

Comment:
To avoid path separator issues - use the `path` module :wink:. You can do stuff like:

```ts
const path = path.resolve(__dirname, 'somewhere', 'else') // absolute-path-to-current-dir/somewhere/else
```

Check out [the docs](https://nodejs.org/api/path.html#path_path_resolve_paths)